lineage/figures/figure9.py

```diff
@@ -9,24 +9,44 @@
 from .figureCommon import getSetup, subplotLabel
 from ..tHMM import tHMM
 
-desired_num_states = np.arange(1, 5)
+desired_num_states = np.arange(1, 8)
+Ts = []
+PIs = []
 
+data = [Lapatinib_Control + Gemcitabine_Control, Lapt25uM, Lapt50uM, Lap250uM, Gem5uM, Gem10uM, Gem30uM]
+
+# to find the T and pi matrices to be used as the constant and reduce the number of estimations.
+for i in desired_num_states:
+    tHMM_solver = tHMM(X=Gemcitabine_Control, num_states=i)
+    tHMM_solver.fit()
+    # choose the estimated shape parameters for 1-state model to be kept constant
+    if i == 1:
+        constant_shape = [int(tHMM_solver.estimate.E[0].params[2]), int(tHMM_solver.estimate.E[0].params[4])]
+    for _ in range(len(data)):
+        Ts.append(tHMM_solver.estimate.T)
+        PIs.append(tHMM_solver.estimate.pi)
+
+for population in data:
+    for st in desired_num_states:
+        thmm_sol = tHMM(X=population, num_states=st)
```

Comment:
You are setting to a new tHMM object you then never use... I'll fix this right now.